buster/chatbot.py

```diff
@@ -149,53 +156,48 @@ def prepare_prompt(
         documents_str: str = self.prepare_documents(matched_documents, max_words=self.cfg.max_words)
         return text_before_documents + documents_str + text_before_prompt + question
 
-    def get_gpt_response(self, **completion_kwargs):
+    def get_gpt_response(self, **completion_kwargs) -> Response:
         # Call the API to generate a response
         logger.info(f"querying GPT...")
         try:
-            return openai.Completion.create(**completion_kwargs)
-
+            response = openai.Completion.create(**completion_kwargs)
         except Exception as e:
             # log the error and return a generic response instead.
             logger.exception("Error connecting to OpenAI API. See traceback:")
-            response = {"choices": [{"text": "We're having trouble connecting to OpenAI right now... Try again soon!"}]}
-            return response
+            return Response("", True, "We're having trouble connecting to OpenAI right now... Try again soon!")
+
+        text = response["choices"][0]["text"]
+        return Response(text)
 
-    def generate_response(self, prompt: str, matched_documents: pd.DataFrame, unknown_prompt: str) -> str:
+    def generate_response(
+        self, prompt: str, matched_documents: pd.DataFrame, unknown_prompt: str
+    ) -> tuple[Response, Iterable[Source]]:
         """
         Generate a response based on the retrieved documents.
         """
         if len(matched_documents) == 0:
             # No matching documents were retrieved, return
-            return unknown_prompt
+            return Response(unknown_prompt), tuple()
```

Comment:
a bit nitpicky here, but could we do 

```
sources = tuple()
return Response(unknown_prompt), sources
```

instead? just so it is explicit what the empty tuple is supposed to represent

Reply:
```suggestion
            sources = tuple()
            return Response(unknown_prompt), sources
```